lesson22/src/js/main.js

```diff
@@ -222,9 +220,17 @@ const resetStatusForSortButton = () => {
     });
 }
 
+const filterSortCategories = () =>{
+    const column = Object.entries(tableColumnConfig).map(([key,value])=>({key, value}));
+    const sortCategories = column.filter((a) => a.value.hasSort );
+    return sortCategories;
+}
+
```

Comment:
I'm using map to turn tableColumnConfig into an array object, and then filtering it, 
but I feel like it's a bit of a pain.
 If you have a better way, please let me know!

I'm thinking that it might be better to make tableColumnConfig an array object from the start.
````Java Script
//map result
{key: "id", {value: 'ID', hasSort: true}},
{key: "name", {value: '名前', hasSort: false}},
{key: "gender", {value: '性別', hasSort: false}},
{key: "age", {value: '年齢', hasSort: true}}
````

Reply:
IMO
If you only need to know if it is a sort category or not, why don't you add a click event in the `createSortButtons` instead of putting in a filter process?
I had the same problem when I implemented it.

Reply:
Thank you for your comment!

I see! You're right, 
that way I don't need to filter the object. 
I changed the method to add a click event in the createSortButtons.

https://github.com/nakagawamai/js-lesson/pull/24/commits/87affd08f535faaa11f8c203a0b371bbf703f573